arviz/plots/lmplot.py

```diff
@@ -0,0 +1,274 @@
+    show=None,
+    figsize=None,
+    textsize=None,
+    axes=None,
+    legend=True,
+    grid=True,
+):
+    """Plot linear regression figure to visualize uncertaininty in mean and ppc values.
+
+    Parameters
+    ----------
+    y : str or DataArray or ndarray
+        If str, variable name from observed_data
+    idata : az.InferenceData object, Optional
+        Optional only if y is Sequence
+    x : str or tuple of strings or DataArray or array-like, Optional
+        If str or tuple, variable name from constant_data
+        If ndarray, could be 1D, or 2D for multiple plots
+        If none, coords name of y (y should be DataArray).
+    y_model : str or Sequence, Optional
+        If str, varibale name from posterior.
+        Its dimensions should be same as y plus added chains and draws.
+    y_ppc : str, Optional
+        If str, varibale name from posterior_predictive.
+        Its dimensions should be same as y plus added chains and draws.
+    num_pp_samples : int, Optional, Default 50
+    kind_pp : ["samples", "hdi"], Optional, Default "samples"
+    kind_model : ["lines", "hdi"], Optional, Default "lines"
+    plot_dim : str, Optional
+        Necessary if y is multidimentional.
+    backend : str, optional
+        Select plotting backend {"matplotlib","bokeh"}. Default "matplotlib".
+    y_kwargs : Passed to plot() in matplotlib and circle() in bokeh
+    y_ppc_plot_kwargs : Passed to plot() in matplotlib and circle() in bokeh
+    y_ppc_fill_kwargs : Passed to az.plot_hdi()
+    y_model_plot_kwargs : Passed to plot() in matplotlib and line() in bokeh
+    y_model_fill_kwargs : Passed to az.plot_hdi()
+    backend_kwargs
+    figsize : tuple, optional
+        Figure size. If None it will be defined automatically.
+    textsize : float, optional
+        Text size scaling factor for labels, titles and lines. If None it will be
+        autoscaled based on figsize.
+    axes : numpy array-like of matplotlib axes or bokeh figures, optional
+        A 2D array of locations into which to plot the densities. If not supplied, Arviz will create
+        its own array of plot areas (and return it).
+    show: bool, optional
+        Call backend show function.
+    legend : bool, optional
+        Add legend to figure. By default True.
+    grid : bool, optional
+        Add grid to figure. By default True.
+
+
+    Returns
+    -------
+    axes
+    """
+    if kind_pp not in ("samples", "hdi"):
+        raise ValueError("kind_ppc could be either samples or hdi")
+
+    if kind_model not in ("lines", "hdi"):
+        raise ValueError("kind_model could be either lines or hdi")
+
+    if y_ppc is None and isinstance(y, str):
+        y_ppc = y
+
+    if isinstance(y, str):
+        y = idata.observed_data[y]
+    elif not isinstance(y, DataArray):
+        y = xr.DataArray(y)
+
+    if len(y.dims) > 1 and plot_dim is None:
+        raise ValueError("Plot_dim argument is needed in case of multidimensional data")
```

Comment:
```suggestion
        raise ValueError("Argument plot_dim is needed in case of multidimensional data")
```

Make the capitalization more clear